### PR TITLE
Use env var for auth frontend

### PR DESCRIPTION
**Pull request by tarzzz**
- [x] Test on desktop app
  - [x] Test on on-prem

### CODE REVIEW

**Comment by tarzzz** [comment]
@n-riesco Please review..

File: app/components/Configuration.react.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ class Configuration extends Component {
         this.state = {
             isMenuOpen: false,
             username: cookie.load('db-connector-user'),
-            authDisabled: cookie.load('db-connector-auth-disabled')
+            // eslint-disable-next-line no-undef
```
Comment (n-riesco) [inline app/components/Configuration.react.js]: use `/*global PLOTLY_ENV*/` instead

File: webpack.config.web.js [hunk 2]
```diff
@@ -23,6 +23,11 @@ const config = {
         new webpack.DefinePlugin({
             __DEV__: false,
             'process.env.NODE_ENV': JSON.stringify('production')
+        }),
+        new webpack.DefinePlugin({
```
Comment (n-riesco) [inline webpack.config.web.js]: Please, add a comment explaining why this is needed.

**Comment by n-riesco** [comment]
@tarzzz This PR does what you explained in [solution 1](https://github.com/plotly/streambed/issues/10436#issuecomment-351064663). How about @scjody 's [solution](https://github.com/plotly/streambed/issues/10436#issuecomment-351122646)? Wouldn't it be easier and avoid the build step at the ENTRYPOINT?

File: webpack.config.web.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ const config = {
             __DEV__: false,
             'process.env.NODE_ENV': JSON.stringify('production')
         }),
+        /*
+         * This is used to pass environment variables to frontend when docker
+         * container initializes
+         */
```
Comment (n-riesco) [inline webpack.config.web.js]: Sorry for being a pain, but this is not enough. We'll forget very soon. Please, could you at least add a link to the streambed issue?

File: app/components/Configuration.react.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ class Configuration extends Component {
         this.state = {
             isMenuOpen: false,
             username: cookie.load('db-connector-user'),
-            // eslint-disable-next-line no-undef
+            /*global PLOTLY_ENV*/
```
Comment (n-riesco) [inline app/components/Configuration.react.js]: linter fails here, spacing needed: `/* global PLOTLY_ENV */`

File: webpack.config.web.js [hunk 5]
```diff
@@ -2,6 +2,9 @@ import webpack from 'webpack';
 import baseConfig from './webpack.config.base';
 import path from 'path';
 
+const AUTH_ENABLED = (process.env.PLOTLY_CONNECTOR_AUTH_ENABLED !== 'undefined' ?
```
Comment (n-riesco) [inline webpack.config.web.js]: This doesn't work. How about:
```
const AUTH_ENABLED = process.env.PLOTLY_CONNECTOR_AUTH_ENABLED ?
    JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED) :
    true;
```
?

**Comment by tarzzz** [comment]
@n-riesco Addressed the comments.. Please let me know if it looks good.

I have also added a `OAUTH2_CLIENT_ID` variable, which is to replace the hardcoded `clientId` in `Login.react.js`. Apart from that, this is also needed for another on-premise issue.

File: webpack.config.web.js [hunk 6]
```diff
@@ -2,8 +2,12 @@ import webpack from 'webpack';
 import baseConfig from './webpack.config.base';
 import path from 'path';
 
-const AUTH_ENABLED = (process.env.PLOTLY_CONNECTOR_AUTH_ENABLED !== 'undefined' ?
-                      JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED): true);
+const AUTH_ENABLED = process.env.PLOTLY_CONNECTOR_AUTH_ENABLED ?
+                     JSON.parse(process.env.PLOTLY_CONNECTOR_AUTH_ENABLED) : true;
```
Comment (n-riesco) [inline webpack.config.web.js]: :cow: the rest of the code uses 4-space indent

File: webpack.config.web.js [hunk 7]
```diff
@@ -2,8 +2,12 @@ import webpack from 'webpack';
...
+
+const OAUTH2_CLIENT_ID = process.env.PLOTLY_CONNECTOR_OAUTH2_CLIENT_ID ?
+                         JSON.stringify(process.env.PLOTLY_CONNECTOR_OAUTH2_CLIENT_ID) :
+                         JSON.stringify('isFcew9naom2f1khSiMeAtzuOvHXHuLwhPsM7oPt');
```
Comment (n-riesco) [inline webpack.config.web.js]: :cow: the rest of the code uses 4-space indent

**Comment by n-riesco** [comment]
This PR is OK as a quick fix for the onprem release, but we need to find an alternative solution straight after release.

@scjody , are you OK with having the cliend ID stored as a global for this quick fix?

**Comment by tarzzz** [comment]
> are you OK with having the cliend ID stored as a global for this quick fix?

More info, this is related to consent screen bug (https://github.com/plotly/streambed/issues/10444), where we can provide custom client ID as an On Prem Environment Variable. For non-prem, the client ID will still point to https://plot.ly/

**Comment by scjody** [comment]
> are you OK with having the cliend ID stored as a global for this quick fix?

@n-riesco It's stored as a global in the current version. What's your new concern here?

**Comment by n-riesco** [comment]
@scjody it's stored in local variable [here](https://github.com/plotly/falcon-sql-client/blob/0568b760aac7f8cfb372d20bb70729c63894ada1/app/components/Login.react.js#L129). I may be wrong, but I don't think Falcon uses global variables (by global variable I mean a variable defined in the global context: `window` in the browser and `global` in Node).

**Comment by scjody** [comment]
Seems reasonable to me. It's not a confidential value; anyone who looks at this repo can see it. (OAuth security in our case depends on the `redirect_uri` being something that an attacker can't serve data from, not from keeping client parameters secret.)

**Comment by tarzzz** [comment]
Addressed the 🐮 ..

@n-riesco 💃 ??

**Comment by n-riesco** [comment]
:dancer:

**Pull request merged**